### PR TITLE
Weather forecast request

### PR DESCRIPTION
**Pull request by yarostbaklajana**
The purpose of this pull request is :
-to make request to OpenWeatherMap site 
-to handle response 
-to create array with forecast cards for 3-day period
-to print to console array with forecast cards

### CODE REVIEW

File: src/components/app.jsx [hunk 1]
```diff
@@ -1,21 +1,95 @@
...
+    async handleSubmit(cityID) {
+        try {
+            console.log(await this.getForecasts(cityID));
+        } catch (err) {
+            console.error(err);
+        }
+        
+    }
+
+    async getForecasts(cityID) {
```
Comment (DarkXaHTeP) [inline src/components/app.jsx]: It is rather forecast than forecasts

File: src/components/app.jsx [hunk 2]
```diff
@@ -1,21 +1,95 @@
...
+import moment from 'moment';
 
 class App extends React.Component {
     constructor(props) {
         super(props);
 
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    async handleSubmit(cityID) {
```
Comment (DarkXaHTeP) [inline src/components/app.jsx]: Use camelCase for arguments names everywhere in the code. "ID" is NOT an abbreviation

File: src/components/app.jsx [hunk 3]
```diff
@@ -1,21 +1,95 @@
...
+        try {
+            const fiveDayForecast = await axios.get('http://api.openweathermap.org/data/2.5/forecast', {
+                params: {
+                    id: cityID,
+                    APPID: process.env.OPENWEATHER_API_KEY,
+                    units: 'metric'
+                }
+            });
+            
+            const threeDayForecastList = fiveDayForecast.data.list.slice(0, 21);
```
Comment (DarkXaHTeP) [inline src/components/app.jsx]: 21 (items in your list for three days) x 3 (new item in list every three hours) = 63 (hours of forecast)
63 / 3 (days required) = 21 (amounts of hours in your imaginary day)

File: src/components/app.jsx [hunk 4]
```diff
@@ -1,21 +1,95 @@
...
+
+            return this.getForecastCards(threeDayForecastList);
+        } catch (err) {
+            throw err.message;
+        }
+    }
+
+    getForecastCards(allForecastedDataArray) {
+        let forecastCards = [];
```
Comment (DarkXaHTeP) [inline src/components/app.jsx]: Why not const?

File: src/components/app.jsx [hunk 5]
```diff
@@ -1,21 +1,95 @@
...
+        allForecastedDataArray.forEach((singleForecast) => { forecastCards.push({
+                dateTime: this.convertToLocaleDateTime(singleForecast.dt),
+                weather: singleForecast.weather[0].description,
+                weatherIconId: singleForecast.weather[0].id,
```
Comment (DarkXaHTeP) [inline src/components/app.jsx]: Why don't you insert complete icon URL for convenience? This icon id is useless without whole URL anyway

File: src/components/app.jsx [hunk 6]
```diff
@@ -1,21 +1,95 @@
...
+            return forecastCards;
+    }
+
+    convertToLocaleDateTime(utc) {
+        const dateTime = moment.unix(utc);
+        return dateTime.format('DD/MM/YYYY - HH:mm'); 
+    }
+
+    convertHectopascalsToMmHg(hPa) {
+        return hpa * 0,75;
```
Comment (DarkXaHTeP) [inline src/components/app.jsx]: will always return 0

File: src/components/app.jsx [hunk 7]
```diff
@@ -1,21 +1,95 @@
...
+    }
+
+    convertToCompassDirection(directionInDeg) {
+        if (directionInDeg >= 337.2 && directionInDeg <= 22.5) {
```
Comment (DarkXaHTeP) [inline src/components/app.jsx]: Why 337.2?

**Review by DarkXaHTeP — CHANGES_REQUESTED** [review]

File: src/components/app.jsx (hunk 7 above)
Comment (yarostbaklajana) [reply src/components/app.jsx]: It's misprint

File: src/components/app.jsx [hunk 8]
```diff
@@ -10,26 +10,26 @@ class App extends React.Component {
...
                 params: {
-                    id: cityID,
+                    id: cityId,
                     APPID: process.env.OPENWEATHER_API_KEY,
                     units: 'metric'
                 }
             });
             
-            const threeDayForecastList = fiveDayForecast.data.list.slice(0, 21);
+            const threeDayForecastList = fiveDayForecast.data.list.slice(0, 25);
```
Comment (DarkXaHTeP) [inline src/components/app.jsx]: Make sure that date-time is in Ukrainian local time and adjust number of cards to not exceed three days

File: src/components/app.jsx [hunk 9]
```diff
@@ -38,14 +38,14 @@ class App extends React.Component {
...
 
     getForecastCards(allForecastedDataArray) {
-        let forecastCards = [];
+        const forecastCards = [];
         allForecastedDataArray.forEach((singleForecast) => { forecastCards.push({
                 dateTime: this.convertToLocaleDateTime(singleForecast.dt),
                 weather: singleForecast.weather[0].description,
-                weatherIconId: singleForecast.weather[0].id,
+                weatherIconUrl: `http://openweathermap.org/img/w/${singleForecast.weather[0].id}.png`,
                 temperature: singleForecast.main.temp,
```
Comment (DarkXaHTeP) [inline src/components/app.jsx]: `temperature: -2.68` as far as I can see on the screenshot provided in App description, the format of the temperature is different

File: src/components/app.jsx [hunk 10]
```diff
@@ -60,11 +60,11 @@ class App extends React.Component {
     }
 
     convertHectopascalsToMmHg(hPa) {
-        return hpa * 0,75;
+        return (hPa * 0.75).toFixed(0);
```
Comment (DarkXaHTeP) [inline src/components/app.jsx]: Use Math.round

**Review by DarkXaHTeP — CHANGES_REQUESTED** [review]

**Review by DarkXaHTeP — APPROVED** [review]

**Pull request merged**